### PR TITLE
Add bearer token support for token server requests

### PR DESCRIPTION
**Pull request by woigl**
I've added the support of a bearer token for the requests to the token server. I require this enhancement, as my backend server requires authentication/authorization to proceed with the requests.

I would appreciate it if you could pull that enhancement into your codebase and publish a new library/npm package version.

### CODE REVIEW

**Comment by tasoskakour** [comment]
Hey @woigl :wave:  
Thanks for the PR! 

Couple of notes: 
- I propose instead of passing a `exchangeCodeForTokenAccessToken` to just pass a generic headers object like `exchangeCodeForTokenHeaders`. In this way we support all kind of headers that someone would want to send to their backend :slightly_smiling_face: 
- Looks like you applied some formatting to the README.md. I think this isn't needed at this point.

**Comment by woigl** [comment]
Hi @tasoskakour,

Thanks for you feedback.

Will do the following:
- That's a great idea because `exchangeCodeForTokenHeaders` will make it an even more flexible approach.
- That was not intentional, my prettier interfered here. 

Will you close this PR and I will generate a new one?

**Comment by woigl** [comment]
Hi @tasoskakour,

I've reverted my initial changes enhanced as proposed by you.

Let me know if you want me to do more on that.

Thanks!

File: README.md [hunk 1]
```diff
@@ -122,6 +122,7 @@ This is the hook that makes this package to work. `Options` is an object that co
 - **extraQueryParameters** (string - _optional_): An object of extra parameters that you'd like to pass to the query part of the authorizeUrl, e.g {audience: "xyz"}.
 - **exchangeCodeForTokenServerURL** (string): This property is only used when using _code authorization grant_ flow (responseType = code). It specifies the API URL of your server that will get called immediately after the user completes the authorization flow. Read more [here](#what-is-the-purpose-of-exchangecodefortokenserverurl-for-authorization-code-flows).
 - **exchangeCodeForTokenMethod** (string - _optional_): Specifies the HTTP method that will be used for the code-for-token exchange to your server. Defaults to **POST**.
+- **exchangeCodeForTokenHeaders** (string - _optional_): An object of extra parameters that will be used for the code-for-token exchange to your server.
```
Comment (tasoskakour) [inline README.md:125]: `object - _optional_` here

File: src/components/use-oauth2.test.ts [hunk 2]
```diff
@@ -169,7 +169,7 @@ describe('useOAuth2', () => {
 					REDIRECT_URI,
 					generatedState as string
 				),
-				{ method: 'POST' },
+				{ method: 'POST', headers: {} },
```
Comment (tasoskakour) [inline src/components/use-oauth2.test.ts:172]: It would be nice to send some actual headers here and verify from server side that they are being sent :)

**Review by tasoskakour — COMMENTED** [review]
Nice one! Left a couple of comments :slightly_smiling_face:

File: README.md (hunk 1 above)
Comment (woigl) [reply README.md]: I've changed from string to object.
As this also seems wrong for `extraQueryParameters`, therefore I've changed that too.

File: src/components/use-oauth2.test.ts (hunk 2 above)
Comment (woigl) [reply src/components/use-oauth2.test.ts]: I've added a bearer token (just taken the sample token from https://jwt.io/) for the comparison of the result accordingly.

**Comment by woigl** [comment]
Thanks for the comments ;-)

All done.

**Comment by tasoskakour** [comment]
Nice one! Merged :heavy_check_mark:

**Pull request merged**